test/functional/blockprod_generate_pos_blocks.py

```diff
@@ -25,6 +25,8 @@
 
 import random, time
 
+COIN = 100_000_000_000
```

Comment:
This should be a global definition somewhere. I bet there's already a definition somewhere for bitcoin. We shouldn't redefine it for every test.